packages/devtools_app/lib/src/performance/flutter/performance_screen.dart

```diff
@@ -29,82 +40,105 @@ class PerformanceScreen extends Screen {
+  final PerformanceController controller = PerformanceController();
 
   @override
-  PerformanceBodyState createState() => PerformanceBodyState();
-}
-
-class PerformanceBodyState extends State<PerformanceBody> {
-  final PerformanceController _controller = PerformanceController();
-
-  CpuProfileData _data;
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            _buildStateControls(),
+            ProfileGranularityDropdown(),
+          ],
+        ),
+        Expanded(
+          child: ValueListenableBuilder(
+            valueListenable: controller.cpuProfilerController.dataNotifier,
+            builder: (context, cpuProfileData, _) {
+              if (cpuProfileData ==
+                  controller.cpuProfilerController.dummyEmptyCpuProfileData) {
+                return _buildRecordingInfo();
+              }
+              return _buildCpuProfiler(cpuProfileData);
+            },
+          ),
+        ),
+      ],
+    );
+  }
 
-  @override
-  void initState() {
-    super.initState();
-    // TODO(djshuckerow): add in buttons to control the CPU recording.
-    _controller.startRecording();
-    Future.delayed(const Duration(seconds: 3)).then((_) async {
-      if (!mounted) return;
-      await _controller.stopRecording();
-      _controller.cpuProfileTransformer.processData(_controller.cpuProfileData);
-      setState(() {
-        // Note: it's not really clear what the source of truth for data is.
-        // We're copying a value out of the controller and storing it in
-        // this state. There's no real reason to not just use it directly
-        // from the controller. We also want a way of making sure that
-        // the controller doesn't change this value without an update to this
-        // State instance.
-        _data = _controller.cpuProfileData;
-        // TODO(djshuckerow): remove when this screen includes buttons to
-        // expand/collapse all by default.
-        _data.cpuProfileRoot.expandCascading();
-      });
-    });
+  Widget _buildStateControls() {
+    const double minIncludeTextWidth = 600;
+    return ValueListenableBuilder(
+      valueListenable: controller.recordingNotifier,
+      builder: (context, recording, _) {
+        return Row(
+          children: [
+            ...recordStopButtons(
+              recordKey: PerformanceScreen.recordButtonKey,
+              stopKey: PerformanceScreen.stopRecordingButtonKey,
+              minIncludeTextWidth: minIncludeTextWidth,
+              recording: recording,
+              onRecord: _startRecording,
+              onStop: _stopRecording,
+            ),
+            const SizedBox(width: 8.0),
+            clearButton(
+              key: PerformanceScreen.clearButtonKey,
+              minIncludeTextWidth: minIncludeTextWidth,
+              onPressed: recording
```

Comment:
The following will be the same:

```dart
onPressed: recording ? null : _clear,
```

`_clear` already is a `Future<void> Function()` so you can pass it as a tearoff.

Reply:
done